Introduce `ngt::AnyBuffer`

fwyzard opened the pull request:
#### PR description:

`ngt::AnyBuffer` behaves like `std::any`, with two differences: it can only be used with trivially copyable types, and provides access to the underlying memory buffer to allow `memcpy`'ing its content.

fwyzard commented:
please test

cmsbuild commented:
cms-bot internal usage<!-- bot cache: {"commits":{"846c960a5720f95399e30ac91124a1e3ad1dc9ed":{"files":["DataFormats/Common/interface/AnyBuffer.h"],"squashed":false,"time":1761667895}},"emoji":{"3457592521":"+1"},"last_seen_sha":"846c960a5720f95399e30ac91124a1e3ad1dc9ed","signatures":{}} -->

fwyzard commented:
OK, understood, though we would appreciate not having to rebuild about 1000 packages every time 🤷🏻‍♂️

fwyzard commented:
Will be moved to `ngt::AnyBuffer` under `HeterogeneousCore/SerialisationCore`.

fwyzard committed:
Introduce ngt::AnyBuffer

ngt::AnyBuffer behaves like std::any, with two differences: it can only be used
with trivially copyable types, and provides access to the underlying memory
buffer to allow memcpy'ing its content.

fwyzard commented:
@cms-sw/core-l2 we are still thinking about the new package name, but let me know if you would prefer to have this under `FWCore` and in the `edm` namespace.

fwyzard commented:
please test
